### ISSUE TITLE
SignalR not working in background fetch

### ISSUE THREAD

**Issue by maniarjmandi** [DyKnow/SignalR-ObjC#219]
hi first excuse me for my bad language i work on local notification with SignalR and i use the background fetch for SignalR . but it work in simulator an do not work and when the app is backgrounded SignalR is not connecting! in my iDevice 4S . this is my code
 
-(void)application:(UIApplication *)application performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {  
    if ([[NSUserDefaults standardUserDefaults]stringForKey:@"UserName"]) {  
        /  
    }  
    completionHandler (UIBackgroundFetchResultNewData);  
}  
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions  
{  
  
    if ([UIApplication instancesRespondToSelector:@selector(registerUserNotificationSettings:)]){  
        [application registerUserNotificationSettings:[UIUserNotificationSettings settingsForTypes:UIUserNotificationTypeAlert|UIUserNotificationTypeBadge|UIUserNotificationTypeSound categories:nil]];  
    }  
    /  
    return YES;  
}  
- (void)performBackgroundTask  
{  
    NSString *buf = [[NSUserDefaults standardUserDefaults]stringForKey:@"UserName"];  
    connection = [SRHubConnection connectionWithURL:@"http:/  
    myHub = [connection createHubProxy:@"MobileNotification"];  
    NSLog(@"%@",myHub);  
    connection.started = ^{  
        [myHub invoke:@"init" withArgs:[NSArray arrayWithObjects: buf, nil]];  
    };  
    connection.reconnected =^{  
        [myHub invoke:@"init" withArgs:[NSArray arrayWithObjects: buf, nil]];  
    };  
    [myHub on:@"agentMessage" perform:self selector:@selector(notificationReceived::)];  
    [connection setDelegate:self];  
    [connection start];  
    [UIApplication sharedApplication].applicationIconBadgeNumber = 0;  
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{  
            });  
  
}  
- (void)notificationReceived:(NSString*)RecipientId:(id)message  
{  
    NSDictionary *json = message;  
    int RId = [[json objectForKey:@"RecipientId"] integerValue];  
    [json objectForKey:@"Username"];  
    NSString *body = [json objectForKey:@"Body"];  
    NSString *time = [json objectForKey:@"CreationTime"];  
  
    UILocalNotification* localNotification = [[UILocalNotification alloc] init];  
    localNotification.fireDate = [NSDate date];  
    localNotification.alertBody = [NSString stringWithFormat:@"%@ - %@",body,time];  
    localNotification.timeZone = [NSTimeZone defaultTimeZone];  
    static int i=1;  
    [UIApplication sharedApplication].applicationIconBadgeNumber = i++;  
    [[UIApplication sharedApplication] scheduleLocalNotification:localNotification];  
    [myHub invoke:@"UpdateStatus" withArgs:[NSArray arrayWithObjects:[NSNumber numberWithInteger:RId],@"Delivered", nil]];  
}  
Please tell me what solution is better for background service and SignalR?
Best Regard Mani

**Comment by maniarjmandi** [comment]
please see this topic on apple https://forums.developer.apple.com/thread/20090

**Comment by maniarjmandi** [comment]
Really ? anyone ?
how can i fix it ?

**Comment by joeldart** [comment]
I would agree with the suggestions you are getting in the forum. Add logging to see where the life cycle is breaking down

**Comment by maniarjmandi** [comment]
2015-11-03 15:25:11.341 VistorTech[2799:773625] Thread <NSThread: 0x15610c90>{number = 1, name = main}:-[SRHubConnection createHubProxy:] [Line 90]
[CONNECTION]    will create proxy MobileNotification
2015-11-03 15:25:11.342 VistorTech[2799:773625] Thread <NSThread: 0x15610c90>{number = 1, name = main}:-[SRConnection negotiate:] [Line 145]
[CONNECTION]    will negotiate
2015-11-03 15:25:11.415 VistorTech[2799:773625] Thread <NSThread: 0x15610c90>{number = 1, name = main}:__26-[SRConnection negotiate:]_block_invoke [Line 153]
[CONNECTION]    negotiation was successful NegotiationResponse: ConnectionId=75f66d57-7916-4e0f-b713-efd90f01c242 ConnectionToken=CphxECYuwE769XDB3en/uI7NlaN9GaScMeyx7OSs9CveM/8DU0/rH+n+MIeytwN4ogGur7NvUn8v8MiN5K4dG3qUA1JkBh2azXnFcfEmovE3Qo81hgOZU9ZyMN1FAh1A Url=/IPA/signalr/signalr ProtocolVersion=1.3.0.0 TryWebSockets=0
2015-11-03 15:25:12.731 VistorTech[2799:773625] Thread <NSThread: 0x15610c90>{number = 1, name = main}:__26-[SRConnection negotiate:]_block_invoke [Line 153]
[CONNECTION]    negotiation was successful NegotiationResponse: ConnectionId=38f93def-5f0d-416b-b5a9-0345356edcf8 ConnectionToken=/P+icCJ5NJe6k/U0DUAdsqDvIVdvTop7KwkI+D4mtZT0AwI3X5k/VVSyeJdvL/R/30TeGXc3NQWcInx/sjZUMOtssdgHEcL+14CDJejEx7F9DqvEQWwst+uSKPJpRBUj Url=/IPA/signalr/signalr ProtocolVersion=1.3.0.0 TryWebSockets=0
2015-11-03 15:25:13.758 VistorTech[2799:773625] Thread <NSThread: 0x15610c90>{number = 1, name = main}:__64-[SRLongPollingTransport poll:connectionData:completionHandler:]_block_invoke [Line 115]
[LONG_POLLING]    LP Receive: {"C":"d-3C068AD6-D,0|T2,0|T3,1","S":1,"M":[]}


this is it ! normally it get the messages but u can see the "M":[] is empty ! it mean the Hub = [connection createHubProxy:@"MobileNotification"]; not working in background fetch . try to find the problem but it too complicate !
